bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyAggregatorServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.properties.impl;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import com.adobe.acs.commons.properties.PropertyAggregatorService;
+import com.day.cq.wcm.api.NameConstants;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = PropertyAggregatorService.class,
+        immediate = true,
+        configurationPolicy = ConfigurationPolicy.REQUIRE)
+@Designate(ocd = PropertyAggregatorServiceImpl.Config.class)
+public class PropertyAggregatorServiceImpl implements PropertyAggregatorService {
+
+    private static final Logger log = LoggerFactory.getLogger(PropertyAggregatorServiceImpl.class);
+
+    private static final String PAGE_PROP_PREFIX = "page_properties";
+    private static final String INHERITED_PAGE_PROP_PREFIX = "inherited_page_properties";
+
+    private List<Pattern> exclusionList;
+    private Map<String, String> additionalData;
+
+    @Override
+    public Map<String, Object> getProperties(final Resource resource) {
```

Comment:
To make this more user friendly, could we return a more author friendly reference to the token?

To an author `page_properties.jcr:title` is going to be a bit cryptic - would be great if they could at least search by "Page Title" and have this token come up in autocomplete, even if it still gets put into the content as `page_properties.jcr:title`

Reply:
You mean in the JavaScript have that type search? There is rudimentary typeahead in that, but the author doesn't see `page_properties.jcr:title` until they actually select from the dropdown of available properties, which do display in a separate way - see screenshot
<img width="555" alt="Screen Shot 2019-06-11 at 11 41 37 AM" src="https://user-images.githubusercontent.com/20479202/59452401-dfd99680-8dca-11e9-98ad-8998c72c57fc.png">